src/Fixer/ControlStructure/NoBreakCommentFixer.php

```diff
@@ -0,0 +1,216 @@
+            if ($tokens[$i]->isGivenKind(array(T_SWITCH, T_IF, T_ELSE, T_ELSEIF, T_FOR, T_FOREACH, T_WHILE, T_DO, T_FUNCTION))) {
+                $empty = false;
+                $i = $this->getStructureEnd($tokens, $i);
+
+                continue;
+            }
+
+            if ($tokens[$i]->isGivenKind(array(T_BREAK, T_CONTINUE, T_RETURN))) {
+                if (null !== $commentPosition) {
+                    $this->removeComment($tokens, $commentPosition);
+                }
+
+                break;
+            }
+
+            if ($tokens[$i]->isGivenKind(array(T_COMMENT)) && '// no break' === $tokens[$i]->getContent()) {
+                $commentPosition = $i;
+                continue;
+            }
+
+            if ($tokens[$i]->isGivenKind(array(T_CASE, T_DEFAULT))) {
+                if (!$empty) {
+                    if (null === $commentPosition) {
+                        $this->insertCommentAt($tokens, $i);
+                    } else {
+                        $this->ensureNewLineAt($tokens, $commentPosition);
+                    }
+                } elseif (null !== $commentPosition) {
+                    $this->removeComment($tokens, $commentPosition);
+                }
+
+                break;
+            }
+
+            if (!$tokens[$i]->isGivenKind(array(T_COMMENT, T_WHITESPACE))) {
+                $empty = false;
+                if (null !== $commentPosition) {
+                    $commentPosition = null;
+                }
+            }
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $casePosition
+     */
+    private function insertCommentAt(Tokens $tokens, $casePosition)
+    {
+        $casePosition += $this->ensureNewLineAt($tokens, $casePosition);
+
+        $tokens->insertAt($casePosition - 1, new Token(array(T_COMMENT, '// no break')));
+
+        $this->ensureNewLineAt($tokens, $casePosition - 1);
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $position
+     *
+     * @return int Number of created tokens
+     */
+    private function ensureNewLineAt(Tokens $tokens, $position)
+    {
+        $lineEnding = $this->whitespacesConfig->getLineEnding();
+        $content = $lineEnding.$this->getIndentAt($tokens, $position);
+
+        $whitespaceToken = $tokens[$position - 1];
+        if (!$whitespaceToken->isGivenKind(T_WHITESPACE)) {
+            $tokens->insertAt($position, new Token(array(T_WHITESPACE, $content)));
+
+            return 1;
+        }
+
+        if (false === strpos($whitespaceToken->getContent(), $lineEnding)) {
+            $whitespaceToken->setContent($content);
+        }
+
+        return 0;
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $commentPosition
+     */
+    private function removeComment(Tokens $tokens, $commentPosition)
+    {
+        $tokens[$commentPosition]->clear();
```

Comment:
I finally used `clearTokenAndMergeSurroundingWhitespace` as it was required for some new test cases I added.

> I think the fixer should not remove the whitespace but leave that to other fixers.

I tend to agree but I think it's better if the fixer is able to leave code as clean as possible on its own instead of assuming other fixers will clean it after.